### PR TITLE
openssh@9.9p1.bcr.1

### PR DESCRIPTION
**Pull request by malt3**
This supersedes #3143 and #3148:

- Prepare a new version of openssh
- Fix include of `crypt.h` (now with a minimally invasive config change)
- Package for macOS
  - The `@openssh//:ssh` and `@openssh//:sshd` targets were tested on Intel and Apple Silicon

### CODE REVIEW

File: modules/openssh/9.9p1.bcr.1/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+matrix:
+  bazel:
+    - 7.x
+    - 8.0.0rc2
+  platform:
```
Comment (mattyclarkson) [inline modules/openssh/9.9p1.bcr.1/presubmit.yml:5]: Add MacOS builds? We could filter `sshd` or just build a subset on MacOS until we get `@libxcrypt` on MacOS?

Comment (mattyclarkson) [reply modules/openssh/9.9p1.bcr.1/presubmit.yml:5]: Or we wait for #3150, update the dependency and do the full build.

Comment (malt3) [reply modules/openssh/9.9p1.bcr.1/presubmit.yml:5]: Yes. Let’s merge libxcrypt and then upgrade the dep and add the CI check.

**Comment by mattyclarkson** [comment]
LGTM, thanks!

**Comment by shixiong-jiang** [comment]
Hello @malt3,

One simple question plz.

My `MODULE.bazel`

```bazel
bazel_dep(name = "openssh", version = "9.9p1")

single_version_override(
    module_name = "openssh",
    patch_strip = 1,
    patches = ["//patches:openssh.patch"],
    version = "9.9p1",
)
``` 

openssh.patch

```
+
+cc_library(
+    name = "ssherr",
+    hdrs = ["ssherr.h"],
+    visibility = ["//visibility:public"],
+)
+
```

My cc. 

It works

```cc
#include "ssherr.h"
```

However, I prefer 

```cc
#include "openssh/ssherr.h"
```

Filename `ssherr.h` is very easy to conflict with other file names.

Thx,

**Comment by malt3** [comment]
Hi @shixiong-jiang, I think this a question for @mattyclarkson, the maintainer.
This would require you to add `include_prefix = "openssh"` to cc_library targets in the OpenSSH module. This can probably be added as a configurable setting, but I‘m not the right person to ask about that.

**Comment by shixiong-jiang** [comment]
@malt3, Thx. 

```
+
+cc_library(
+    name = "ssherr",
+    hdrs = ["ssherr.h"],
+    include_prefix = ["openssh"],
+    visibility = ["//visibility:public"],
+)
+
```

did work 👍

**Pull request merged**